search/searcher/search_term.go

```diff
@@ -60,17 +64,86 @@ func NewTermSearcherBytes(ctx context.Context, indexReader index.IndexReader, te
 	if err != nil {
 		return nil, err
 	}
-	return newTermSearcherFromReader(indexReader, reader, term, field, boost, options)
+	return newTermSearcherFromReader(ctx, indexReader, reader, term, field, boost, options)
 }
 
-func newTermSearcherFromReader(indexReader index.IndexReader, reader index.TermFieldReader,
-	term []byte, field string, boost float64, options search.SearcherOptions) (*TermSearcher, error) {
+func tfTDFScoreMetrics(indexReader index.IndexReader) (uint64, float64, error) {
+	// default tf-idf stats
 	count, err := indexReader.DocCount()
 	if err != nil {
-		_ = reader.Close()
-		return nil, err
+		return 0, 0, err
+	}
+	// field cardinality metric is not used in the tf-idf scoring algo.
+	fieldCardinality := 0
+	if count == 0 {
+		return 0, 0, nil
+	}
+	return count, float64(fieldCardinality / int(count)), nil
```

Comment:
It seems the 2nd parameter is always 0 here, so why declare it even and do a divide by operation and then return it. Change return params to 0 and `avgDocLength` below where this method is referenced will remain at zero.